server/utils.go

```diff
@@ -174,3 +185,196 @@ func (w *withCode) Format(s fmt.State, verb rune) {
 		}
 	}
 }
+
+func fetchListOfResources(clusterType string, environment string, loginURL string, tenantId string) (map[string][]map[string]map[string]oputil.DataAction, error) {
+	operationsMap := map[string][]map[string]map[string]oputil.DataAction{}
+
+	apiResourcesList, err := fetchApiResources()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to fetch list of api-resources from apiserver.")
+	}
+
+	operationsList, err := fetchDataActionsList(environment, clusterType, loginURL, tenantId)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to fetch operations from Azure.")
+	}
+
+	for _, resList := range apiResourcesList {
+		group := "v1" // core api group
+		if resList.GroupVersion != "" && resList.GroupVersion != "v1" {
+			group = strings.Split(resList.GroupVersion, "/")[0]
+		}
+
+		if len(resList.APIResources) == 0 {
+			continue
+		}
+
+		for _, apiResource := range resList.APIResources {
+			if strings.Contains(apiResource.Name, "/") {
+				continue
+			}
+
+			actionId := clusterType
+			if group != "v1" {
+				actionId = path.Join(actionId, group)
+			}
+
+			resourceName := apiResource.Name
+
+			actionId = path.Join(actionId, resourceName)
+
+			for _, operation := range operationsList {
```

Comment:
i think you can convert the operationsList into a map of sort so that you don't need to keep looping them over and over again